mods/bucket/init.lua

```diff
@@ -166,18 +171,21 @@ minetest.register_craftitem("bucket:bucket_empty", {
 			local source_neighbor = false
 			if liquiddef.force_renew then
 				source_neighbor =
-					minetest.find_node_near(pointed_thing.under, 1, liquiddef.source)
+					minetest.find_node_near(pos, 1, liquiddef.source)
 			end
-			if not (source_neighbor and liquiddef.force_renew) then
-				minetest.add_node(pointed_thing.under, {name = "air"})
+			if source_neighbor and liquiddef.force_renew then
+				log_action(pos, pname, "picked up " .. liquiddef.source .. " (force renewed)")
+			else
+				minetest.add_node(pos, {name = "air"})
```

Comment:
unrelated but this could be `minetest.remove_node(pos)`

Reply:
This exists before this PR... Is it good to fix it in this PR?

Reply:
> This exists before this PR... Is it good to fix it in this PR?

Yes, generally you're free to fix some style issues as you go - just don't bloat the scope of the PR (e.g. only fix style issues in code you're actually touching, the rest is better left to cleanup PRs).

Reply:
There's actually a second occurrence of this in MTG so this is better done in a separate commit.